lib/list_versions_test.go

```diff
@@ -215,96 +213,117 @@ func TestRemoveDuplicateVersions(t *testing.T) {
 	var version string
-	version = "0.11.8"
 
+	// Test valid version formats
+	version = "0.11.8"
 	valid := validVersionFormat(version)
-
 	if valid == true {
 		t.Logf("Valid version format : %s (expected)", version)
 	} else {
 		t.Errorf("Failed to verify version format: %s\n", version)
 	}
 
 	version = "1.11.9"
-
 	valid = validVersionFormat(version)
-
 	if valid == true {
 		t.Logf("Valid version format : %s (expected)", version)
 	} else {
 		t.Errorf("Failed to verify version format: %s\n", version)
 	}
 
-	version = "1.11.a"
-
+	version = "1.11.9-beta1"
 	valid = validVersionFormat(version)
-
-	if valid == false {
-		t.Logf("Invalid version format : %s (expected)", version)
+	if valid == true {
+		t.Logf("Valid version format : %s (expected)", version)
 	} else {
 		t.Errorf("Failed to verify version format: %s\n", version)
 	}
 
-	version = "22323"
-
+	version = "0.12.0-rc2"
 	valid = validVersionFormat(version)
+	if valid == true {
+		t.Logf("Valid version format : %s (expected)", version)
+	} else {
+		t.Errorf("Failed to verify version format: %s\n", version)
+	}
 
-	if valid == false {
-		t.Logf("Invalid version format : %s (expected)", version)
+	version = "1.11.4-boom"
+	valid = validVersionFormat(version)
+	if valid == true {
+		t.Logf("Valid version format : %s (expected)", version)
 	} else {
 		t.Errorf("Failed to verify version format: %s\n", version)
 	}
 
-	version = "@^&*!)!"
+	// Test valid minor version format
+	version = "1.11"
+	valid = validVersionFormat(version, regexSemVer.Minor)
+	if valid == true {
+		t.Logf("Valid minor version format : %s (expected)", version)
+	} else {
+		t.Errorf("Failed to verify minor version format: %s\n", version)
+	}
 
-	valid = validVersionFormat(version)
+	// Test valid patch version format
+	version = "1.11.4"
+	valid = validVersionFormat(version, regexSemVer.Patch)
+	if valid == true {
+		t.Logf("Valid patch version format : %s (expected)", version)
+	} else {
+		t.Errorf("Failed to verify patch version format: %s\n", version)
+	}
 
+	// Test invalid version formats
+	version = "1.11.a"
+	valid = validVersionFormat(version)
 	if valid == false {
 		t.Logf("Invalid version format : %s (expected)", version)
 	} else {
 		t.Errorf("Failed to verify version format: %s\n", version)
 	}
 
-	version = "1.11.9-beta1"
-
+	version = "22323"
 	valid = validVersionFormat(version)
-
-	if valid == true {
-		t.Logf("Valid version format : %s (expected)", version)
+	if valid == false {
+		t.Logf("Invalid version format : %s (expected)", version)
 	} else {
 		t.Errorf("Failed to verify version format: %s\n", version)
 	}
 
-	version = "0.12.0-rc2"
-
+	version = "@^&*!)!"
 	valid = validVersionFormat(version)
-
-	if valid == true {
-		t.Logf("Valid version format : %s (expected)", version)
+	if valid == false {
+		t.Logf("Invalid version format : %s (expected)", version)
 	} else {
 		t.Errorf("Failed to verify version format: %s\n", version)
 	}
 
-	version = "1.11.4-boom"
-
+	version = "1.11.4-01"
```

Comment:
Why is : `version = "0.12.0-rc2"` valid, but this isn't? 🤔

Reply:
https://semver.org/#spec-item-9 (see the text highlighted with bold font)
> A pre-release version MAY be denoted by appending a hyphen and a series of dot separated identifiers immediately following the patch version. Identifiers MUST comprise only ASCII alphanumerics and hyphens [0-9A-Za-z-]. Identifiers MUST NOT be empty. **Numeric identifiers MUST NOT include leading zeroes**. […]